kahuna/public/js/image/controller.js

```diff
@@ -208,7 +208,18 @@ image.controller('ImageCtrl', [
       return ctrl.image.data.source.dimensions;
     }
 
-    mediaCropper.getCropsFor(image).then(crops => {
+    mediaCropper.getCropsFor(image).then(cropsResource => {
+      let crops = cropsResource.data;
+      if ($window._clientConfig.canDownloadCrop) {
+        crops.forEach((crop) => {
+          crop.assets.forEach((asset) =>
+            asset.downloadLink = cropsResource.links.find(link => link.rel.includes(`crop-download-${crop.id}-${asset.dimensions.width}`))?.href
+          );
+          //set the download link of the crop to be the largest asset
+          let largestAsset = crop.assets.find(asset => asset.dimensions.width == crop.master.dimensions.width);
```

Comment:
this is the logic for finding the 'extreme' assets (i.e. smallest and largest) that I'd hinted at in my previous comment... https://github.com/guardian/grid/blob/ca7fa005e870fe9ddf0a18aabc80e511bedd891a/kahuna/public/js/main.js#L316-L328
...then example usage (of smallest)...
https://github.com/guardian/grid/blob/e669dfc4b01aa4fdfa82893d535e4a86558319b3/kahuna/public/js/image/crop.html#L4 

Totally understand if they're not suitable here.

Reply:
Yes, I saw this logic, however this function takes images as input and I needed to take crops instead, so I thought I would just rewrite it instead of trying to generalize this function only for this use case. Also getExtremeAssets is slightly more inefficient since it needs to sort out the assets array, whereas I only need to find the asset with the original crop width.